keystorage/interlock/src/main/java/tech/pegasys/signers/interlock/InterlockSession.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.signers.interlock;
+
+import java.nio.file.Path;
+
+public interface InterlockSession extends AutoCloseable {
+
+  /**
+   * Fetch key from given path. It is expected that the private key is stored in hex format in given
+   * file.
+   *
+   * @param keyPath The path of key file in Interlock, for instance "/bls/key1.txt"
+   * @return contents from file defined by path.
+   * @throws InterlockClientException In case of an error while fetching key
+   */
+  String fetchKey(Path keyPath) throws InterlockClientException;
```

Comment:
Is the key required to be in hexadecimal? If it is then we could just return Bytes

Reply:
I think its a safe assumption for the contents to be in hex specially for "signer" scenario. It would have been converted to Bytes in Web3Signer layer anyways, converting it here would eliminate that step. Have pushed refactored changes to cover it.